samplekotlin/src/main/java/com/example/chattutorial/ChannelActivity.kt

```diff
@@ -59,27 +58,19 @@ class ChannelActivity : AppCompatActivity(R.layout.activity_channel) {
         }
 
         // step 6 - handle back button behaviour correctly when you're in a thread
-        val backButtonHandler = {
+        channelHeaderView.onBackClick = {
             messageListViewModel.onEvent(MessageListViewModel.Event.BackButtonPressed)
         }
-        channelHeaderView.onBackClick = { backButtonHandler() }
 
-        onBackPressedDispatcher.addCallback(
-            this,
-            object : OnBackPressedCallback(true) {
-                override fun handleOnBackPressed() {
-                    backButtonHandler()
-                }
-            }
-        )
+        onBackPressedDispatcher.addCallback(this) {
+            channelHeaderView.onBackClick()
+        }
     }
 
     companion object {
         private const val CID_KEY = "key:cid"
 
         fun newIntent(context: Context, channel: Channel) =
-            Intent(context, ChannelActivity::class.java).apply {
-                putExtra(CID_KEY, channel.cid)
-            }
+            Intent(context, ChannelActivity::class.java).putExtra(CID_KEY, channel.cid)
```

Comment:
While we're here, this two-line one-liner method might as well have a regular body instead of an expression body. Or at the very least, an explicit return type.

Reply:
True, I do generally like explicit return types on methods with expression bodies, though this one is named reasonably explicitly.